### ISSUE TITLE
Add an `empty` condition to function spacing rules

### ISSUE THREAD

**Issue by dkrutsko** [eslint/eslint#7829]
* **ESLint Version:** 3.12.2
* **Node Version:** 6.9.2
* **NPM Version:** 4.0.5
* **Parser** babel-eslint

I'd like to propose an addition to the `space-before-function-paren` and `func-call-spacing` rules (along with others I might be missing). A setting of `empty` (or another name) would only enforce spaces when the function has parameters. As an example, the following code would cause an error:

```js
alert('Hello');
console.log(42);
new Date ();
functionCall ();

function withoutSpace(x) {
}

var anonymousWithSpace = function () {
};
```

While the following would be correct:
```js
alert ('Hello');
console.log (42);
new Date();
functionCall();

function withSpace (x) {
}

var anonymousWithoutSpace = function() {
};
```

In summary, called or declared functions would require a space before the opening parens if it has parameters, however, if there are no parameters then there should not be a space.

**Comment by nzakas** [comment]
Thanks for the suggestion. I'm not sure this option is general enough to include in the core rule, as I'm having a hard time understanding why an empty function should obey different rules. 

I'd suggest creating a custom rule if you need this for your codebase.

**Comment by platinumazure** [comment]
@nzakas I think here, "empty" refers to the argument list. In other words, `console.log(arg)` is good, but OP wants `foo ()` to have a space. Maybe a better proposal would be to have always/never for "withArgs" and "withoutArgs" scenarios. That said, I think I agree with your ultimate conclusion.

**Comment by nzakas** [comment]
Looks like there isn't support on the team for this enhancement, so closing.

**Comment by dkrutsko** [comment]
I'm sorry to hear that but thank you for taking my suggestion under consideration :-)
